internal/hash.go

```diff
@@ -21,38 +34,78 @@ var algos = map[string]Hasher{
 
 var RecommendedAlgo = "sha256"
 
+// Hasher defines a function that returns a hash.Hash.
```

Comment:
These extra comments are helpful, but they're not really part of the functionality of this PR.  It would probably be good to remove them from this PR and put them in a separate PR.